test/workflows/components/params.libsonnet

```diff
@@ -0,0 +1,17 @@
+{
```

Comment:
I think we are moving away from ksonnet for KF components installation, not for our prow testing infra. The infra is set up using ks and unlikely to change in the near future.

Reply:
Thanks for setting this up @zhenghuiwang. However, I think I'm missing something here. Are we using ksonnet to set up a Kubeflow cluster or something else? For testing right now, we have no dependencies on anything. So, we could just as well use prow to run unit tests for now. We probably want to run integration tests in a Kubeflow cluster, but in that case, do we need ksonnet? 

I'd like to avoid adding unnecessary complexity to the test infra and only add things when we need them. What do you think?

Reply:
https://github.com/kubeflow/testing#adding-an-e2e-test-to-a-repository

We are leveraging the e2e testing infra, which assumes there is a ksonnet app that defines workflows. The KS app.yaml points to the testing cluster. There are some initial setup in [run_e2e_workflow.py](https://github.com/kubeflow/testing/blob/master/py/kubeflow/testing/run_e2e_workflow.py), such as copying the source code and setting up the ENV.

I had the same concern but it seem to be the best option for setting up right now, especially if we are going to add e2e tests in the future. I've tried to trim down unnecessary KS code to make it clear.

Reply:
@jlewi what do you think?

Reply:
@neuromage Our testing infra has a lot of advantages; in particular the presubmit/postsubmit tests are defined in the repo itself rather than in a separate repository.

This makes modifying fixing the tests much easier; e.g. if you want to add a new test you can add the test in a PR and the presubmit will actually invoke the test and ensure it passes.

Our test infrastructure was designed to kick off Argo workflows. We used Argo so that we could define E2E tests as DAGs. We treat unittests as one step workflows.

Our Argo workflows are parameterized to easily allow running different versions of a test (e.g. with istio vs. without).

We happen to use ksonnet as a way of parameterizing Argo workflows.

The primary reason to use ksonnet to express the workflows is because its the path of least resistance. If you want to support some non ksonnet way of doing it; you'd need to modify some of the test infra.

In this context ksonnet is only used by Kubeflow developers so there is not as much rush to migrate off it.

One proposal is to use KFP to write E2E tests; but that requires some investement. 

So until then its simpler to just write tests using ksonnet.

Reply:
Thanks @jlewi and @zhenghuiwang for the explanation. In pipelines, we also use Argo workflows, but the only use of ksonnet is as a CLI call to deploy a Kubeflow cluster. I guess I'm unsure as to when we would need to parameterize those workflows in the metadata case, and hence was asking if we really need this flexibility right now given the complexity cost and the fact that ksonnet was going away. But I'm ok with doing it this way if this is the path of least resistance to get our test infra going.

Thanks again for setting this up!

/lgtm

Reply:
Thanks for the reviews!